Entity Dump broken on Windows

### General Info

Terasology Launcher Version: 4.8.0
Terasology Version: 5.3.0
Operating System: Windows 11
Onboard / Dedicated Graphics: Dedicated
Java Version: Launcher-bundled

### What you were trying to do

Dump entities from the in-game console of a Terasology world, running on Windows.

### What actually happened

`An error occurred while executing command 'dumpEntities': Illegal char <:> at index 13: 2023-09-27T20:52:11.811415100Z-entityDump.json`

### How to reproduce

1. Select Terasology version '5.3.0'
1. Start Terasology
1. Select gameplay 'CoreGameplay' (any)
1. Start game
1. Open in-game console (F1)
1. Execute `dumpEntities`
1. See error

### Log details

```
22:52:11.815 [main] ERROR o.t.engine.logic.console.ConsoleImpl - An error occurred while executing a command
org.terasology.engine.logic.console.commandSystem.exceptions.CommandExecutionException: java.nio.file.InvalidPathException: Illegal char <:> at index 13: 2023-09-27T20:52:11.811415100Z-entityDump.json
	at org.terasology.engine.logic.console.commandSystem.AbstractCommand.execute(AbstractCommand.java:229)
	at org.terasology.engine.logic.console.ConsoleImpl.execute(ConsoleImpl.java:251)
[...]
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 13: 2023-09-27T20:52:11.811415100Z-entityDump.json
	at java.base/sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPath.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.resolve(Unknown Source)
	at org.terasology.engine.logic.console.commands.CoreCommands.dumpEntities(CoreCommands.java:482)
```

### Workaround

Use Linux 🤪

--- a/engine/src/main/java/org/terasology/engine/logic/console/commands/CoreCommands.java
+++ b/engine/src/main/java/org/terasology/engine/logic/console/commands/CoreCommands.java
@@ -479,7 +479,7 @@ public class CoreCommands extends BaseComponentSystem {
         PrefabSerializer prefabSerializer =
                 new PrefabSerializer(engineEntityManager.getComponentLibrary(), engineEntityManager.getTypeSerializerLibrary());
         WorldDumper worldDumper = new WorldDumper(engineEntityManager, prefabSerializer);
-        Path outFile = PathManager.getInstance().getHomePath().resolve(Instant.now() + "-entityDump.json");
+        Path outFile = PathManager.getInstance().getHomePath().resolve(Instant.now().toString().replaceAll(":", "-") + "-entityDump.json");
         if (componentNames.length == 0) {
             savedEntityCount = worldDumper.save(outFile);
         } else {

